### PR TITLE
ENH: remove global vars in adapters

### LINKED ISSUE #352: ENH: remove global vars in adapters
### Is your feature request related to a problem? Please describe
Currently we are using global vars to store method fallbacks.

For example,
https://github.com/xprobe-inc/xorbits/blob/7c86482bf61619a053f0078cb52a6bea40225f0d/python/xorbits/numpy/__init__.py#L74

### Describe the solution you'd like
Generally, global vars should be avoided. We can cache the return values of
https://github.com/xprobe-inc/xorbits/blob/7c86482bf61619a053f0078cb52a6bea40225f0d/python/xorbits/numpy/numpy_adapters/core.py#L74
so that the global vars could be removed.

For the way to cache a functions' results, you can refer to https://docs.python.org/3/library/functools.html#functools.lru_cache

### PR DESCRIPTION
**Pull request by jiayini1119**
## Related issue number

Fixes #352 

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/numpy/numpy_adapters/core.py [hunk 1]
```diff
@@ -71,6 +72,7 @@ def _get_output_type(func: Callable) -> MarsOutputType:
         return _NO_ANNOTATION_FUNCS.get(func, MarsOutputType.object)
 
 
+@lru_cache(maxsize=1)
 def collect_numpy_module_members(np_mod: ModuleType) -> Dict[str, Any]:
```
Comment (qianduoduo0904) [inline python/xorbits/numpy/numpy_adapters/core.py:76]: This method is already called `collect_numpy_module_members`, do we still need to pass the numpy module into it?

Comment (jiayini1119) [reply python/xorbits/numpy/numpy_adapters/core.py:76]: hi, thanks for your review! I found that currently, we are passing different numpy modules (```numpy```, ```numpy.fft```, ```numpy.linalg``` etc) to this function to generate the fallback methods in their init.py. Should I modify this logic? Thanks.

Comment (qianduoduo0904) [reply python/xorbits/numpy/numpy_adapters/core.py:76]: Oh, it's OK when passing submodules, but just curious that the functions for numpy & pandas act differently.

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

**Comment by aresnow1** [comment]
Could you rebase main branch and resolve the conflicts?

**Comment by qianduoduo0904** [comment]
Some conflict needs to be addressed, other parts look great to me.

**Review by UranusSeven — APPROVED** [review]
LGTM

**Pull request merged**